source/distributed/tria.cc

```diff
@@ -2866,31 +1935,17 @@ namespace parallel
           Assert(false, ExcInternalError());
         }
 
-      // load saved data, if any was stored
-      if (cell_attached_data.n_attached_deserialize > 0)
-        {
-          data_transfer.load(parallel_forest,
-                             filename,
-                             attached_count_fixed,
-                             attached_count_variable);
-
-          data_transfer.unpack_cell_status(local_quadrant_cell_relations);
-
-          // the CellStatus of all stored cells should always be CELL_PERSIST.
-          for (const auto &quad_cell_rel : local_quadrant_cell_relations)
-            {
-              (void)quad_cell_rel;
-              Assert(
-                (std::get<1>(quad_cell_rel) == // cell_status
-                 parallel::distributed::Triangulation<dim,
-                                                      spacedim>::CELL_PERSIST),
-                ExcInternalError());
-            }
-        }
-
       this->update_periodic_face_map();
       this->update_number_cache();
 
+      // Load attached cell data, if any was stored.
+      this->load_attached_data(parallel_forest->global_first_quadrant[myrank],
+                               parallel_forest->global_num_quadrants,
+                               parallel_forest->local_num_quadrants,
+                               filename,
+                               attached_count_fixed,
+                               attached_count_variable);
+
```

Comment:
Why do you load attached data after updating the cache now?

Reply:
This was an oversight. Fixed, thanks! 😉